### PR TITLE
Edit cpp enum

### LINKED ISSUE #5483: [Edit] C++ enum
### Type of Edit (select all that apply)

Add new content (definitions, codeblocks, etc.)

### Description (optional)

We would like to edit the `enum` concept entry under C++. The entry lives in `content/cpp/concepts/enum/enum.md`

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) guide when working on the PR for this issue.

Please:

1. Shorten the description to not more than 160 characters
2. Fix any typo in the entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by RashmitTopG**
### Description

Made modifications in cpp enum . Added codeblocks and definitions.

### Issue Solved

Closes #5483

Made modifications in cpp enum . Added codeblocks and definitions.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [✓] All writings are my own.
- [✓] My entry follows the Codecademy Docs style guide.
- [✓] My changes generate no new warnings.
- [✓] I have performed a self-review of my own writing and code.
- [✓] I have checked my entry and corrected any misspellings.
- [✓] I have made corresponding changes to the documentation if needed.
- [✓] I have confirmed my changes are not being pushed from my forked `main` branch.
- [✓] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [✓] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/cpp/concepts/enum/enum.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 Title: 'Enums'
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:2]:
```suggestion
Title: 'Enum'
```

File: content/cpp/concepts/enum/enum.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-Description: 'In C++, an enumeration (enum) is a user defined type where a set of values are specified for a variable and the variable can only take one out of a small set of possible values.'
+Description: 'In C++, an enum defines a variable with a limited set of predefined values.'
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:3]:
```suggestion
Description: 'Defines a variable with a limited set of predefined values.'
```

File: content/cpp/concepts/enum/enum.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
 Subjects:
   - 'Computer Science'
   - 'Game Development'
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:6]: Please add tags and catalog content

File: content/cpp/concepts/enum/enum.md [hunk 4]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
 
 if (logger_result == LogResult::Success) {} // Because Success is scoped to LogResult, it doesn't collide with SocketResult::Success
 ```
+
+## Enum to Int Conversion
+
+In C++, enums can be implicitly converted to integers, useful for numeric contexts like array indexing or bitwise operations.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:108]:
```suggestion
In C++, `enum` can be implicitly converted to integers, useful for numeric contexts like array indexing or bitwise operations.
```

File: content/cpp/concepts/enum/enum.md [hunk 5]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+
+```cpp
+enum color { red, green, blue };
+
+int main() {
+    color c = green;
+    int colorValue = c;  // Implicit conversion to int
+    std::cout << "Color value: " << colorValue; // Output: Color value: 1
+}
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:117]:
```suggestion
#include <iostream>
enum color { red, green, blue };

int main() {
    color c = green;
    int colorValue = c;  // Implicit conversion to int
    std::cout << "Color value: " << colorValue; // Output: Color value: 1
}
```

File: content/cpp/concepts/enum/enum.md [hunk 6]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+```
+
+Converting an enum to int is easy, but converting int to enum is risky as no bounds check is done, leading to undefined behavior if the value is out of range.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:120]:
```suggestion
Converting an `enum` to `int` is easy, but converting `int` to `enum` is risky as no bounds check is done, leading to undefined behavior if the value is out of range.
```

File: content/cpp/concepts/enum/enum.md [hunk 7]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+
+## Custom Underlying Type for Enums
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:122]:
```suggestion
## Custom underlying type for Enums
```

File: content/cpp/concepts/enum/enum.md [hunk 8]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+
+By default, an enum's type is int, but you can specify a smaller type like unsigned char to optimize memory usage.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:124]:
```suggestion
By default, an enum's type is `int`, but you can specify a smaller type like unsigned `char` to optimize memory usage.
```

File: content/cpp/concepts/enum/enum.md [hunk 9]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+enum class Permission : unsigned char {
+    Read = 1,
+    Write = 2,
+    Execute = 4
+};
+
+int main() {
+    Permission p = Permission::Write;
+    std::cout << static_cast<int>(p);  // Explicit cast to int: Output: 2
+}
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:136]:
```suggestion
#include <iostream>
enum class Permission : unsigned char {
    Read = 1,
    Write = 2,
    Execute = 4
};

int main() {
    Permission p = Permission::Write;
    std::cout << static_cast<int>(p);  // Explicit cast to int: Output: 2
}
```

File: content/cpp/concepts/enum/enum.md [hunk 10]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+```
+
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:138]: Please add output for the above code

File: content/cpp/concepts/enum/enum.md [hunk 11]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+Here, the underlying type of Permission is unsigned char. The constants Read, Write, and Execute are stored using only 1 byte of memory.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:139]:
```suggestion
Here, the underlying type of Permission is `unsigned char`. The constants `Read`, `Write`, and `Execute` are stored using only 1 byte of memory.
```

File: content/cpp/concepts/enum/enum.md [hunk 12]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+
+## Best Practices for Using Enums
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:141]:
```suggestion
## Best practices for using Enums
```

File: content/cpp/concepts/enum/enum.md [hunk 13]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+
+1. Use enum class for Strong Typing: Scoped enums (C++11) prevent implicit int conversions, ensuring better type safety.
+2. Explicit Casting: Use static_cast<int>(enum_value) for safe conversions.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:144]:
```suggestion
2. Explicit Casting: Use `static_cast<int>(enum_value)` for safe conversions.
```

File: content/cpp/concepts/enum/enum.md [hunk 14]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+    Permission p = Permission::Write;
+    std::cout << static_cast<int>(p);  // Explicit cast to int: Output: 2
+}
+```
+
+Here, the underlying type of Permission is unsigned char. The constants Read, Write, and Execute are stored using only 1 byte of memory.
+
+## Best Practices for Using Enums
+
+1. Use enum class for Strong Typing: Scoped enums (C++11) prevent implicit int conversions, ensuring better type safety.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:143]:
```suggestion
1. Use enum class for strong typing: Scoped enums (C++11) prevent implicit int conversions, ensuring better type safety.
```

File: content/cpp/concepts/enum/enum.md [hunk 15]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+2. Explicit Casting: Use static_cast<int>(enum_value) for safe conversions.
+3. Avoid Magic Numbers: Enums replace hardcoded numbers, improving readability.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:145]:
```suggestion
3. Avoid magic numbers: Enums replace hardcoded numbers, improving readability.
```

File: content/cpp/concepts/enum/enum.md [hunk 16]
```diff
@@ -102,3 +102,45 @@ LogResult logger_result = LogResult::Success;
...
+4. Use Underlying Types Wisely: Choose the underlying type carefully in memory-constrained environments.
```
Comment (Radhika-okhade) [inline content/cpp/concepts/enum/enum.md:146]:
```suggestion
4. Use underlying types wisely: Choose the underlying type carefully in memory-constrained environments.
```

**Comment by Radhika-okhade** [comment]
Hey! @RashmitTopG 
I have suggested a few changes. Could you please go through them?

**Review by Radhika-okhade — APPROVED** [review]

**Review by Sriparno08 — APPROVED** [review]
Looks good, @RashmitTopG!

**Pull request merged**